### ISSUE TITLE
[Python] Generates `.fs.py` when `outDir` option is not provided

### ISSUE THREAD

**Issue by MangelMaxime** [fable-compiler/Fable#3616]
### Description

Hello @dbrattli,

The python target seems to be the only to not generates `.fs.py` when `outDir` is not provided.

Is there a reason for that or this is just something forgotten / not known at the time ?

It would be nice if we could generate `.fs.py` files as it allows IDE/Editors like VSCode to nest the files together to not pollute too much the explorer.

![CleanShot 2023-11-27 at 15 33 16@2x](https://github.com/fable-compiler/Fable/assets/4760796/ef4af0df-b067-4bfa-9077-ebca1c350d2a)

**Comment by dbrattli** [comment]
Yes, files are modules in Python so files cannot contain dots in their name, since that would Python think there's a submodule when importing them i.e that the name before the dot is a folder and the name after the dot is a file. https://peps.python.org/pep-0008/#package-and-module-names

**Comment by dbrattli** [comment]
Using _fs.py would be an option. I didn't think about it at the time. But I suppose it will still not be grouped like above?

**Comment by MangelMaxime** [comment]
> But I suppose it will still not be grouped like above?

It will in VSCode, you can configure the rule for nesting the file, in this case using the following works:

```json
"*.fs": "${basename}_fs.py",
```

If we do implement this feature, I will send a update to Ionide as it is missing others nesting rules.

**Comment by smoothdeveloper** [comment]
@MangelMaxime, the main issue with this, as I understand it, is that it precludes authoring python libraries in F#.

all the modules will have _fs suffix for python consumers.

**Comment by MangelMaxime** [comment]
@smoothdeveloper all the modules will have _fs suffix for python consumers.

Not necessarily, if you provide `--outDir` or `-e py` then the generated files with have `.py` extension. This is the same behaviour as JavaScript target.

**Comment by smoothdeveloper** [comment]
@dbrattli, I'm not yet sensibilized with all the ways python works with loading modules, I think people actually update `sys.path` or `%PYTHONPATH%` when needing to load modules from separate locations.

I tried `poetry` but the thing generated some attributes in my pyproject.toml, then it made `pip install .` fail for sake of missing readme.md file with the whole backtrace displayed...

I'd be worried about forcing people to install fable packages (in site-packages) by default (dll hell), or to force usage of `poetry` by default (it is probably nice but I think people using it know how to configure it without need for fable to do anything).

I'd rather see a FAQ with guidance how to update `sys.path`, pointers to python official documentation around the topic, and maybe some extra pointers on how to work in achieving the same with `poetry`, but not as something we force nor recommend, at least for now.

If python users adopt fable, and start suggesting better integration with `poetry`, then we'd consider what this looks like, but I doubt this should remove the "you can make a fable (or any python) code base file system layout work by handling `sys.paths`, like is highlighted in the official docs" approach, with a small example and ELI5 comments.

Does that sound reasonable? I'm not a python expert, and for now, loading fable generated code and the fable standard library in both a structure of scripts (in same tree) or jupyter notebook worked without doing anything (I have a pyproject.toml and __init__.py files at the right locations though).
